### ISSUE TITLE
On M Error, Output All 5 Error Fields

### ISSUE THREAD

**Issue by bgribaudo** [microsoft/vscode-powerquery-sdk#134]
### Preflight Checklist

- [x] I have installed the [latest version of Power Query SDK](https://github.com/microsoft/vscode-powerquery-sdk/releases/latest).
- [X] I have checked existing resources, including the [common issues](https://learn.microsoft.com/en-us/power-query/commonissues) and the [release notes](https://github.com/microsoft/vscode-powerquery-sdk/releases).
- [X] I have searched for [similar issues](https://github.com/microsoft/vscode-powerquery-sdk/issues).

### Problem

Currently, `PQTest.exe run-test` only outputs an M error's message

### Desired Solution

`PQTest.exe run-test` to output all 5 of an error's fields (Reason, Message, Detail, Message.Format, Message.Parameters), and possibly the location metadata

### Additional Context

These additional details can help with debugging (both when a dev. runs a test file from VSCode, as well as when wading through CI/CD logs after the fact).

**Comment by mattmasson** [comment]
I was working on a change to expose Reason and Details.

Do you see value in exposing Message.Format and Message.Parameters as well? I'm not sure how they would be used.

**Comment by bgribaudo** [comment]
> Do you see value in exposing Message.Format and Message.Parameters as well? I'm not sure how they would be used.

I could see Message.Format being helpful if PQTest was being used in an automation scenario to run data quality validation tests against live, in-production data. If an error occurs, it may be desirable to log the error without any PII that its message potentially contains—something that Message.Format would be useful for.

On the other hand, if PII wasn't present or logging it wasn't a concern, being able to log the message template and its parameters separately makes it easier to do error analysis (see the "summarize errors" scenario given in https://bengribaudo.com/blog/2022/05/24/6753/new-m-feature-structured-error-messages).

**Comment by mattmasson** [comment]
This is resolved with the latest SDK Tools update - 2.110.3.

Something like: 

```powerquery
error Error.Record("DataSource.Error", "parameterized '#{0}' with number '#{1}'", [ detail = "hello", field = 1], {"p1", 1})
```
Would show up in a new `Error` element in the evaluation result. 

```json
{
        "Error": {
            "Message": "parameterized 'p1' with number '1'",
            "MessageFormat": "parameterized '{0}' with number '{1}'",
            "MessageParameters": [
                "p1",
                "1"
            ],
            "Details": {
                "Reason": "DataSource.Error",
                "detail": "hello",
                "field": 1
            }
}
```

I decided to include `Reason` as part of the details, since that is how it comes back in the underlying API. It simplified things, because other types of exceptions (like the ones you get for invalid expressions) will contain additional fields in the Details section (like location).

**Comment by bgribaudo** [comment]
Thanks!
